src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VQSRTrainingSetManager.java

```diff
@@ -0,0 +1,144 @@
+package org.broadinstitute.hellbender.tools.walkers.vqsr;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class is used to manage access to training sets in VQSR.
+ */
+public final class VQSRTrainingSetManager {
+
+    private final List<DataSet> dataSets;
+
+    public VQSRTrainingSetManager() {
+        this.dataSets = new ArrayList<>();
+    }
+
+    /*
+     * Helper class to represent a data set used in VQSR.
+     */
+    private final static class DataSet {
+        private final FeatureInput<VariantContext> fi;
+        private final boolean isKnown;
+        private final boolean isTraining;
+        private final boolean isAntiTraining;
+        private final boolean isTruth;
+        private final boolean isConsensus;
+        private final double prior;
+
+        private final static Logger logger = LogManager.getLogger(DataSet.class);
+
+        public DataSet(final FeatureInput<VariantContext> fi) {
+            this.fi = fi;
+
+            final String name = fi.getName();
+
+            // Parse the tags to decide which tracks have which properties
+            isKnown = Objects.equals(fi.getAttribute("known"), "true");
+            isTraining = Objects.equals(fi.getAttribute("training"), "true");
+            isAntiTraining = Objects.equals(fi.getAttribute("bad"), "true");
+            isTruth = Objects.equals(fi.getAttribute("truth"), "true");
+            isConsensus = Objects.equals(fi.getAttribute("consensus"), "true");
+            prior = ( fi.getAttribute("prior") != null ? Double.parseDouble(fi.getAttribute("prior")) : 0.0 );
+
+            // Report back to the user which tracks were found and the properties that were detected
+            if( !isConsensus && !isAntiTraining ) {
+                logger.info( String.format("Found %s track: \tKnown = %s \tTraining = %s \tTruth = %s \tPrior = Q%.1f", name, isKnown, isTraining, isTruth, prior) );
+            } else if( isConsensus ) {
+                logger.info( String.format("Found consensus track: %s", name) );
+            } else {
+                logger.info( String.format("Found bad sites training track: %s", name) );
+            }
+        }
+
+        /**
+         * Annotates the VariantDatum with information from this data set.
+         */
+        public void annotateDatum(FeatureContext tracker, VariantContext evalVC, VariantDatum datum, boolean trust_all_polymorphic) {
```

Comment:
This method seems like it should potentially be a method on VariantDatum instead of here.